src/rdkafka_cgrp.c

```diff
@@ -3774,46 +4102,60 @@ rd_kafka_consumer_group_metadata_new (const char *group_id) {
         rd_kafka_consumer_group_metadata_t *cgmetadata;
 
         cgmetadata = rd_kafka_consumer_group_metadata_new_with_genid(group_id,
-                                                                     -1);
+                                                                     -1, "",
+                                                                     NULL);
 
         return cgmetadata;
 }
 
 rd_kafka_consumer_group_metadata_t *
 rd_kafka_consumer_group_metadata_new_with_genid (const char *group_id,
-                                                 int32_t generation_id) {
+                                                 int32_t generation_id,
+                                                 const char *member_id,
+                                                 const char
+                                                 *group_instance_id) {
         rd_kafka_consumer_group_metadata_t *cgmetadata;
 
-        if (!group_id)
-                return NULL;
+        rd_assert(group_id);
```

Comment:
We typically don't check the validity of input arguments unless we need it to catch bugs, but this will crash three lines down anyway if any of these are null, and the calling conventions are clear, so not really necessary and just causes confusion: "Why is this checked here when other places dont check arguments?"

Reply:
sounds good.